miner/worker.go

```diff
@@ -1117,7 +1141,12 @@ func (w *worker) prepareWork(genParams *generateParams) (*environment, error) {
 	}
 	// Run the consensus preparation with the default or customized consensus engine.
 	if err := w.engine.Prepare(w.chain, header); err != nil {
-		log.Error("Failed to prepare header for sealing", "err", err)
+		switch err.(type) {
```

Comment:
Could we use errors.Is(As) here?

Reply:
This comes from cherry-picking the commit, but we can try `errors.Is` here.